gpt_index/indices/knowledge_graph/base.py

```diff
@@ -73,6 +75,9 @@ def __init__(
             **kwargs,
         )
 
+        if self._index_struct is not None:
+            self._index_struct._mode = mode
```

Comment:
why do we have this? this means that when the knowledge graph is loaded from disk, it'll by default be set to the `MODE_KEYWORDS` mode unless the user explicitly specifies something else; the mode stored in the saved knowledge graph will always be overridden.

if we want the mode to persist in disk (which i'm also not sure about, see my below suggestion), then i think we could probably try removing this logic

Reply:
I think this was a consequence of how I was doing some testing (loading some old indexes), whoops. I agree, it shouldn't be overriding the mode saved in the index